runelite-client/src/main/java/net/runelite/client/ui/overlay/OverlayRenderer.java

```diff
@@ -283,11 +293,8 @@ private Dimension safeRender(RenderableEntity entity, Graphics2D graphics, Point
 		return dimension;
 	}
 
-	private boolean shouldDrawOverlay(Client client, Overlay overlay)
+	private boolean shouldDrawOverlay(Overlay overlay, OverlayLayer layer)
```

Comment:
This method is redundant now, entire logic can be moved directly to the Stream#filter